palette/src/convert.rs

```diff
@@ -499,6 +501,60 @@ where
     }
 }
 
+///A trait for converting from one color to another.
+///
+///This trait wraps the underlying `From` implementation.
+pub trait Convert: Sized {
+    ///Lossless, same as calling T::from(self)
+    #[inline]
+    fn convert<T: From<Self>>(self) -> T {
+        T::from(self)
+    }
+
+    ///Convert into T with values clamped to the color defined bounds
+    fn convert_clamped<T: From<Self> + Limited>(self) -> T {
+        let mut this: T = self.convert();
+        if !this.is_valid() {
+            this.clamp_self();
+        }
+        this
+    }
+
+    ///Convert into Option<T>, returning None if the resulting Color is outside of its defined range
+    fn try_convert<T: From<Self> + Limited>(self) -> Option<T> {
+        let this: T = self.convert();
+        if this.is_valid() {
+            Some(this)
+        } else {
+            None
+        }
+    }
+}
+
+impl Convert for Rgb {}
+impl Convert for Luma {}
+impl Convert for Hsl {}
+impl Convert for Hsv {}
+impl Convert for Hwb {}
+impl Convert for Lab {}
+impl Convert for Lch {}
+impl Convert for Xyz {}
+impl Convert for Yxy {}
+impl<C, T> Convert for Alpha<C, T> where C: Convert, T: Component {}
+
+///A trait for converting from one color to an arbitrary type by handing a closure the color's raw components.
+///
+///This trait makes use of the `Pixel` trait which can be unsafe depending on the implementator.
+pub trait ConvertWith<T>: Pixel<T> where T: Component
+{
+    ///Convert `self` to `T` by applying a function over its components
+    fn convert_with<F, Out>(self, f: F) -> Out where F: FnOnce(&[T]) -> Out, Out: Sized {
+        f(Pixel::into_raw_slice(&[self]))
```

Comment:
This could be `as_raw` and the function could take `&self`. Could also use `into_raw`, but that becomes more complicated.

Reply:
Ah yes, I had problems with coming up for a name for this function actually. `as_raw` should work here I suppose, and I didn't take self by reference because I thought of it being a conversion into something else so consuming self would've fit there. It's probably better to take it be ref nevertheless. Any Ideas on the trait name for that?

Reply:
I have no better name suggestion at the moment.